power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/deserializer.hpp

```diff
@@ -364,6 +367,17 @@ class Deserializer {
     using DataByteMeta = std::vector<std::vector<ComponentByteMeta>>;
     using AttributeByteMeta = std::vector<std::pair<std::string_view, std::vector<std::string_view>>>;
 
+    struct BufferView {
+        WritableDataset::Buffer const* buffer{nullptr};
+        Idx idx{0};
+        std::vector<AttributeBuffer<void> const*> const* attribute_order{nullptr};
```

Comment:
```suggestion
        std::vector<AttributeBuffer<void>> attribute_order{};
```

`AttributeBuffer` is quite small, store in value makes more sense rather than all pointer stuff.
We also do not need an additional indirection of `attribute_order`. Just store it by value. If the component is not columnar, it is just an empty `vector`.

Reply:
then the `BufferView` approach is not possible anymore. We currently copy for every individual component and only update the `idx`. Copying the entire attribute buffer makes this quite expensive if there are e.g. 20 columns.

In fact, this was my first implementation: directly copying the `Buffer`

Reply:
20 columns is still  `20 * sizeof(AttributeBuffer) = 640` bytes, which is relatively negligible to the data volume, which is MBs to GBs.

Reply:
but the copy is made for every single component, not only per component type. so if there are 10k components, then there are ~10k copies. It's part of the hotloop

Reply:
Am I misunderstanding here? You only do this at the beginning of `parse_component`, which is valid for all scenarios and element within that component type. So it only happens once per component type for the entire dataset.

Reply:
this is also done for each scenario in the current implementation

Reply:
I think i found a workaround. Let me dig into it first

Reply:
right, sorry, i was confused cause i'm working on 2 different things simultaneously. it is however true for the serializer.

again. I'll probably need to re-start this PR from scratch